sgt-back-end/index.js

```diff
@@ -0,0 +1,127 @@
+// eslint-disable-next-line no-unused-vars
+const { Pool, Client } = require('pg');
+const express = require('express');
+const app = express();
+
+// only create ONE pool for your whole server
+const db = new Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('listening on port 3000');
+});
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    SELECT *
+    FROM "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'sorry but an unexpected error occurred'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  if (!req.body.score || req.body.score < 0 || req.body.score > 100) {
+    res.status(400).json({ error: 'request must include a valid score between 0 and 100' });
+    return;
```

Comment:
- A `score` of `0` should be treated as a valid score.
- A `score` that is a number, but not an integer, should be treated as invalid.